include/RAJA/pattern/forall.hpp

```diff
@@ -245,13 +270,21 @@ RAJA_INLINE void forall(ExecPolicy<SegmentIterPolicy, SegmentExecPolicy>,
                         const TypedIndexSet<SegmentTypes...>& iset,
                         LoopBody loop_body)
 {
+  util::PluginContext context{util::make_context<ExecPolicy<SegmentIterPolicy, SegmentExecPolicy>>()};
+  util::callPreCapturePlugins(context);
 
   using RAJA::internal::trigger_updates_before;
   auto body = trigger_updates_before(loop_body);
 
+  util::callPostCapturePlugins(context);
+
+  util::callPreLaunchPlugins(context);
+
   wrap::forall(SegmentIterPolicy(), iset, [=](int segID) {
     iset.segmentCall(segID, detail::CallForall{}, SegmentExecPolicy(), body);
   });
+
+  util::callPostLaunchPlugins(context);
```

Comment:
If this is calling the above wrap::forall do we really need to copy the body for each segment?
I think I see why the plugins were being called below instead of here, but should the launch plugins be called for each segment?

Reply:
I've moved the copy of the loop_body into the middle plugin layer so the body should now only be copied once.